paasta_tools/kubernetes_tools.py

```diff
@@ -3359,14 +3359,18 @@ def create_secret(
     secret_data: Dict[str, str],
     namespace: str,
 ) -> None:
+    """
+    :param secret_name: Use _get_secret_name() to generate a secret name
+    :param service_name: Labels have 63 character limit
+    """
     kube_client.core.create_namespaced_secret(
         namespace=namespace,
         body=V1Secret(
             metadata=V1ObjectMeta(
                 name=secret_name,
                 labels={
-                    "yelp.com/paasta_service": service_name,
-                    "paasta.yelp.com/service": service_name,
+                    "yelp.com/paasta_service": sanitise_label_name(service_name),
```

Comment:
i think we only really want to limit the length here - there's no restrictions on characters inside the labels

Reply:
If I don't truncate I get
```
metadata.labels: Invalid value: \"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa\": must be no more than 63 characters
```
from diff
```
--- a/paasta_tools/kubernetes_tools.py
+++ b/paasta_tools/kubernetes_tools.py
@@ -3369,7 +3369,7 @@ def create_secret(
             metadata=V1ObjectMeta(
                 name=secret_name,
                 labels={
-                    "yelp.com/paasta_service": sanitise_label_name(service_name),
+                    "yelp.com/paasta_service": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
```

Reply:
we should still truncate - but https://github.com/Yelp/paasta/pull/3536/files#diff-f1b278fb2675abadd2f58f131c3f67ece5d7511b60fcad691255aeda6f490d17R3739 will do some character replacement which we likely don't want

Reply:
ah, sorry missed that this is the same as what we used to do in the past